### PR TITLE
[Feature][Transform-V2] Support case when Expression

### PR DESCRIPTION
**Pull request by zhilinli123**
### Purpose of this pull request

Support SQL case when syntax

### How was this patch tested?
add sql e2e

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-e2e/seatunnel-transforms-v2-e2e/seatunnel-transforms-v2-e2e-part-2/src/test/resources/sql_transform/case_when.conf [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    ]
+  }
+}
+
+transform {
+  Sql {
+    source_table_name = "fake"
+    result_table_name = "fake1"
+    query = """
+      select case when c_string in ('c_string') then 1 else 0 end     as c_string_1,
```
Comment (EricJoy2048) [inline seatunnel-e2e/seatunnel-transforms-v2-e2e/seatunnel-transforms-v2-e2e-part-2/src/test/resources/sql_transform/case_when.conf:61]: Please add document.

**Review by EricJoy2048 — DISMISSED** [review]
LGTM

Thanks for you contribution. This is a good feature.

**Comment by EricJoy2048** [comment]
@rewerma  PTAL.

File: seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/sql/zeta/ZetaSQLFunction.java [hunk 2]
```diff
@@ -221,6 +227,15 @@ public Object computeForValue(Expression expression, Object[] inputFields) {
             Parenthesis parenthesis = (Parenthesis) expression;
             return computeForValue(parenthesis.getExpression(), inputFields);
         }
+        if (zetaSQLFilter.isConditionExpr(expression)) {
+            return zetaSQLFilter.executeFilter(expression, inputFields);
+        }
```
Comment (Hisoka-X) [inline seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/sql/zeta/ZetaSQLFunction.java]: Why add this? Is we missed something?

Comment (zhilinli123) [reply seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/sql/zeta/ZetaSQLFunction.java]: `case when id = 1 then 0 else 1`
circulation Iterate to the syntax id=1 filter，I have optimized the code writing

**Comment by Hisoka-X** [comment]
CI :https://github.com/zhilinli123/incubator-seatunnel/actions/runs/7550372862/job/20555887983

**Comment by zhilinli123** [comment]
ci fix PTAL: @Hisoka-X @EricJoy2048

**Comment by Hisoka-X** [comment]
image
Some test case failed. Could you retrigger it?

**Comment by zhilinli123** [comment]
> image Some test case failed. Could you retrigger it?

Where is this address? I can't find it.
@Hisoka-X

**Comment by Hisoka-X** [comment]
> Where is this address? I can't find it.

oh ignore it. It already passed.

**Review by Hisoka-X — APPROVED** [review]

**Review by EricJoy2048 — APPROVED** [review]

**Pull request merged**